### PR TITLE
[coq] Adapt to coq/coq#10665

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

File: src/dune [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (library
  (name quickchick_plugin)
  (public_name quickchick.plugin)
- (flags :standard -rectypes -w -8-27)
+ (flags :standard -rectypes -warn-error -3 -w -8-27+40)
```
Comment (SkySkimmer) [inline src/dune]: Is this warning change OK?

Comment (ejgallego) [reply src/dune]: I guess so, otherwise the Dune build does fail with any deprecation warning, which the current tree did have.

Comment (SkySkimmer) [reply src/dune]: I mean are quickchick devs OK with adding warning 40?

Comment (ejgallego) [reply src/dune]: Oh, I thought you were talking about warning 3, I guess it makes sense to add warning 40 too as this is the default in coq_makefile.

Comment (Lysxia) [reply src/dune]: I'm fine with that.

**Comment by SkySkimmer** [comment]
Please merge now

**Pull request merged**